session/nc_creds_sessions_store_test.go

```diff
@@ -0,0 +1,650 @@
+package session_test
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/NordSecurity/nordvpn-linux/config"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/NordSecurity/nordvpn-linux/session"
+	"github.com/NordSecurity/nordvpn-linux/test/category"
+	"github.com/NordSecurity/nordvpn-linux/test/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNCCredentialsSessionStore_Renew_NotExpired(t *testing.T) {
+	category.Set(t, category.Unit)
+
+	uid := int64(123)
+	futureTime := time.Now().UTC().Add(24 * time.Hour)
+
+	cfg := &config.Config{
+		AutoConnectData: config.AutoConnectData{ID: uid},
+		TokensData: map[int64]config.TokenData{
+			uid: {
+				NCData: config.NCData{
+					Username:       "testuser",
+					Password:       "testpass",
+					Endpoint:       "https://api.example.com",
+					ExpirationDate: futureTime,
+				},
+			},
+		},
+	}
+
+	cfgManager := &mock.ConfigManager{Cfg: cfg}
+	errorRegistry := internal.NewErrorHandlingRegistry[error]()
+
+	renewCalled := false
+	renewAPICall := func() (*session.NCCredentialsResponse, error) {
+		renewCalled = true
+		return &session.NCCredentialsResponse{
+			Username:  "newuser",
+			Password:  "newpass",
+			Endpoint:  "https://new.example.com",
+			ExpiresIn: 86400,
+		}, nil
+	}
+
+	store := session.NewNCCredentialsSessionStore(cfgManager, errorRegistry, renewAPICall)
+
+	err := store.Renew()
+	assert.NoError(t, err)
+	assert.False(t, renewCalled, "Renew API should not be called when credentials are valid")
+}
+
+func TestNCCredentialsSessionStore_Renew_NoTokenData(t *testing.T) {
+	category.Set(t, category.Unit)
+
+	uid := int64(123)
```

Comment:
This also seems repeatedly used in the tests. Maybe it's worth (at least consider the effort/gain ratio) to create a simple fixture in here, where all the common stuff could be kept?